File: src/test/kotlin/com/raychenon/kotlin/leetcode/MoveZeroes.kt
package com.raychenon.kotlin.leetcode

/**
 * User: raychenon
 * Date: 17/3/2020
 * https://leetcode.com/problems/move-zeroes/
 */
object MoveZeroes {

    fun moveZeroes(nums: IntArray): Unit {
        var count = nums.toList().filter { x -> x == 0 }.count()

        var temp = 0
        for (i in nums.size - 1..0 step -1) {
            // put only 0 and shift
            if(i >= nums.size - count){
                if (nums.get(i) == 0) {
                    nums[i] = 0
                }else{
                    nums[i] = temp
                    temp = nums.get(i-1)
                }
            }else{

            }

            val value = nums.get(i)
            if (nums.get(i) != 0) {
            }
        }
    }
}

File: python3/algorithms/leetcode/move_zeroes.py
#!/bin/python3
import collections
from typing import List
import re

"""
https://leetcode.com/problems/move-zeroes/
"""
def moveZeroes(nums: List[int]) -> None:
    acc_zeroes_count = 0
    for i in range(len(nums)):
        if nums[i] == 0:
            acc_zeroes_count += 1
        elif acc_zeroes_count > 0:
            temp = nums[i]
            nums[i] = 0
            nums[i-acc_zeroes_count] = temp

if __name__ == "__main__":
    input = [0,1,0,3,12]
    moveZeroes(input)
    assert input == [1,3,12,0,0]